packages/cli/src/commands/init/index.ts

```diff
@@ -0,0 +1,177 @@
+import path from 'path';
+import prompts from 'prompts';
+import type { InitialReturnValue } from 'prompts';
+import { createApp } from './create-app';
+import { validateNpmName } from './helpers/validate-pkg';
+import { isFolderEmpty } from './helpers/is-folder-empty';
+import fs from 'fs';
+import { AnalyticService } from '../../services/analytics.service';
+
+const analytics = new AnalyticService();
+
+const programName = 'novu init';
+
+const onPromptState = (state: { value: InitialReturnValue; aborted: boolean; exited: boolean }) => {
+  if (state.aborted) {
+    /*
+     * If we don't re-enable the terminal cursor before exiting
+     * the program, the cursor will remain hidden
+     */
+    process.stdout.write('\x1B[?25h');
+    process.stdout.write('\n');
+    process.exit(1);
+  }
+};
+
+export interface IInitCommandOptions {
+  secretKey?: string;
+  projectPath?: string;
+  apiUrl: string;
+}
+
+export async function init(program: IInitCommandOptions, anonymousId?: string): Promise<void> {
+  analytics.track({
+    identity: {
+      anonymousId: anonymousId,
+    },
+    data: {},
+    event: 'Run Novu Init Command',
+  });
+
+  let projectPath = program.projectPath;
+
+  if (typeof projectPath === 'string') {
+    projectPath = projectPath.trim();
+  }
+
+  if (!projectPath) {
+    const res = await prompts({
+      onState: onPromptState,
+      type: 'text',
+      name: 'path',
+      message: 'What is your project named?',
+      initial: 'my-novu-app',
+      validate: (name: string) => {
+        const validation = validateNpmName(path.basename(path.resolve(name)));
+        if (validation.valid) {
+          return true;
+        }
+
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        return 'Invalid project name: ' + (validation as any).problems[0];
+      },
+    });
+
+    if (typeof res.path === 'string') {
+      projectPath = res.path.trim();
+    }
+  }
+
+  if (!projectPath) {
+    console.log(
+      '\nPlease specify the project directory:\n' +
+        `  ${cyan(programName)} ${green('<project-directory>')}\n` +
+        'For example:\n' +
+        `  ${cyan(programName)} ${green('my-novu-app')}\n\n` +
+        `Run ${cyan(`${programName} --help`)} to see all options.`
+    );
+    process.exit(1);
+  }
+
+  const resolvedProjectPath = path.resolve(projectPath);
+  const projectName = path.basename(resolvedProjectPath);
+
+  const validation = validateNpmName(projectName);
+  if (!validation.valid) {
+    console.error(`Could not create a project called ${red(`"${projectName}"`)} because of npm naming restrictions:`);
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    (validation as any).problems.forEach((problem: string) => console.error(`    ${red(bold('*'))} ${problem}`));
+    process.exit(1);
+  }
+
+  let userId: string;
+  // if no secret key is supplied set to empty string
+  if (!program.secretKey) {
+    program.secretKey = '';
+  } else {
+    try {
```

Comment:
Small addition to help with telemetry

Reply:
It is not a big improvement but what do you think about adding it above before "What is your project named?" question, fetch it without waiting and here we can await for the response.
in cases of load, it could reduce a couple of milliseconds 😅